crates/sui-json-rpc/src/indexer_api.rs

```diff
@@ -315,79 +324,98 @@ impl<R: ReadApiServer> IndexerApiServer for IndexerApi<R> {
-                            "Cannot find marker field in address info Move struct: {:?}",
-                            address_info_move_struct
+                let domain_bcs_value = bcs::to_bytes(&domain).map_err(|e| {
+                    Error::SuiRpcInputError(SuiRpcInputError::GenericInvalid(format!(
+                        "Unable to serialize name: {:?} with error: {:?}",
+                        domain, e
+                    )))
+                })?;
+                let record_object_id_option = self
+                    .state
+                    .get_dynamic_field_object_id(
+                        registry_id,
+                        name_type_tag,
+                        &domain_bcs_value,
+                    )
+                    .map_err(|e| {
+                        Error::SuiRpcInputError(SuiRpcInputError::GenericInvalid(format!(
+                            "Read name service registry dynamic field table failed with error: {:?}",
+                            e
                         )))
                     })?;
-                let addr = match address_str_move_value {
-                    SuiMoveValue::Address(addr) => Ok(addr),
-                    _ => Err(Error::SuiRpcInputError(SuiRpcInputError::GenericNotFound(
-                        format!("No SuiAddress found in: {:?}", address_str_move_value),
-                    ))),
-                }?;
-                return Ok(Some(addr));
+                if let Some(record_object_id) = record_object_id_option {
+                    let record_object_read =
+                        self.state.get_object_read(&record_object_id).map_err(|e| {
+                            warn!(
+                                "Failed to get object read of name: {:?} with error: {:?}",
+                                record_object_id, e
+                            );
+                            Error::UnexpectedError(format!(
+                                "Failed to get object read of name with error {:?}",
+                                e
+                            ))
+                        })?;
+                    let record_parsed_move_object =
+                        SuiParsedMoveObject::try_from_object_read(record_object_read)?;
+                    // NOTE: "value" is the field name to get the address info
+                    let address_info_move_value = record_parsed_move_object
+                        .read_dynamic_field_value(NAME_SERVICE_VALUE)
+                        .ok_or_else(|| {
+                            Error::UnexpectedError(
+                                "Cannot find value field in record Move struct".to_string(),
+                            )
+                        })?;
+                    let address_info_move_struct = match address_info_move_value {
+                        SuiMoveValue::Struct(a) => Ok(a),
+                        _ => Err(Error::UnexpectedError(
```

Comment:
I think this error might be redundant since `.ok_or_else` already maps None to Err, so we can just convert Value to Struct?

Reply:
this matching is for other types of `SuiMoveValue`
```
    Number(u32),
    Bool(bool),
    Address(SuiAddress),
    Vector(Vec<SuiMoveValue>),
    String(String),
    UID { id: ObjectID },
    Struct(SuiMoveStruct),
    Option(Box<Option<SuiMoveValue>>),
```